### ISSUE TITLE
Hashkey may include colons

### ISSUE THREAD

**Issue by franz-95** [pytest-dev/pytest-html#215]
Hey, my hashkey includes slashes since the tests are in subdirectories. When included in the asset file name, the slashes are wrongly interpreted and in line 161 of the plugin.py the directories are made in the assets directory even though they are part of the assets file name.

https://github.com/pytest-dev/pytest-html/blob/83d6e1bb666ef94a349c5ed5e52babbaba4b1ae3/pytest_html/plugin.py#L158

**Comment by franz-95** [comment]
I just realized that the issue is not the slashes but the colons '::' within the test name. Unfortunately the assets file name cannot include such characters.

**Comment by BeyondEvil** [comment]
Thanks for reporting this @franz-95 !

Mind having a look at this @D3X ?

Looks like it might be related to the changes you made.

Also, I think #214 is related as well. 🤔

**Comment by D3X** [comment]
@BeyondEvil the stuff I did was merely truncating filenames to fit within FS limits. The original work was done in #199 by @SunInJuly :)

It should be easy to fix, though. I suspect we should filer out all potentially problematic characters? Maybe something similar to django's [`slugify`](https://github.com/django/django/blob/febe136d4c3310ec8901abecca3ea5ba2be3952c/django/utils/text.py#L393-L405)?

**Comment by Uil2liv** [comment]
And what about django's [`get_valid_filename`](https://github.com/django/django/blob/febe136d4c3310ec8901abecca3ea5ba2be3952c/django/utils/text.py#L221-L231)?

**Comment by christopheraranda** [comment]
This is also a problem when creating html reports on one platform (Linux) and downloaded to another platform (Windows).

I've redirected all the logging for my tests to a text file and appended it as an extra. The tests run in a CentOS based container, and the html report and assets are compressed to a zip file and emailed out.

Unfortunately, Windows users are unable to decompress the zip file. This is not a problem for Mac and Linux users. I suspect that the forward slashes and colons in the nodeids are having unintended consequences when creating the assets since the tests are contained in subdirectories.

**Comment by BeyondEvil** [comment]
> This is also a problem when creating html reports on one platform (Linux) and downloaded to another platform (Windows).
> 
> I've redirected all the logging for my tests to a text file and appended it as an extra. The tests run in a CentOS based container, and the html report and assets are compressed to a zip file and emailed out.
> 
> Unfortunately, Windows users are unable to decompress the zip file. This is not a problem for Mac and Linux users. I suspect that the forward slashes and colons in the nodeids are having unintended consequences when creating the assets since the tests are contained in subdirectories.

I _think_ this should be its own issue. Mind creating one?

I have a fix, but I'm waiting on a PR review before I can merge it and do a release.

**Comment by christopheraranda** [comment]
@BeyondEvil 
Sure thing. Here's the new [issue](https://github.com/pytest-dev/pytest-html/issues/221).

> > This is also a problem when creating html reports on one platform (Linux) and downloaded to another platform (Windows).
> > I've redirected all the logging for my tests to a text file and appended it as an extra. The tests run in a CentOS based container, and the html report and assets are compressed to a zip file and emailed out.
> > Unfortunately, Windows users are unable to decompress the zip file. This is not a problem for Mac and Linux users. I suspect that the forward slashes and colons in the nodeids are having unintended consequences when creating the assets since the tests are contained in subdirectories.
> 
> @christopheraranda
> I _think_ this should be its own issue. Mind creating one?
> 
> I have a fix, but I'm waiting on a PR review before I can merge it and do a release.

@BeyondEvil 
Sure thing. Here's the new [issue](https://github.com/pytest-dev/pytest-html/issues/221).

**Comment by BeyondEvil** [comment]
@franz-95 

Sorry for the delay addressing this issue.

Would you mind testing out PR #223 and see if that solves the issues you've been seeing? Much appreciated!

**Comment by BeyondEvil** [comment]
Fixed by #223
